src/sknnr/_msn.py

```diff
@@ -67,9 +67,3 @@ class MSNRegressor(YFitMixin, OrdinationKNeighborsRegressor):
 
     def _get_transformer(self) -> TransformerMixin:
         return CCorATransformer(self.n_components)
-
-    def _more_tags(self):
-        return {
-            "multioutput": True,
```

Comment:
@aazuspan, I think I'm leaning toward the suggested fixes you outline here for the tags and, yes, I believe the `CCorATransformer` should match the `CCATransformer`.  Even though it's technically possible to pass the 1d `y` data, I think we want to discourage users from doing that.  My only question to you is whether we want to handle that in a separate PR?  It would be quick and dirty to do here, but perhaps the *rationale* for making that change wouldn't be as clear or might get buried.

Reply:
I think you're right that changing the *content* of the tags should be kept separate from changing the format of the tags, which was the focus of this PR. With that in mind, I'm thinking that I should revert a few changes before merging this. It's a little tricky to keep track since not every tag crosswalks and some are now inherited, but I'm thinking that I should remove the `single_output=False` and `two_d_labels=True` tags from both GNN and CCA, since those weren't previously represented in `_more_tags`. 

Can you double check my logic there?

Reply:
Honestly, I'm slightly confused!  I see that previously in `GNNRegressor`, we had `_more_tags` return `multioutput=True`, but that seems to be different from the new `__sklearn_tags__` defining `single_output=False`?

But I do agree with you that those seem to be new tags that weren't in the previous `_more_tags`.

(And now that I'm digging more, there is more that I don't understand.  For example, the `estimator_type` tag on all of our transformers is `None` when it looks like you can define it as `transformer`.  Then I thought for sure `TransformerMixin` would have an `estimator_type` of `transformer`, but I get this:
```
>>> from sklearn.base import TransformerMixin
>>> print(TransformerMixin().__sklearn_tags__())
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "C:\Users\gregorma\AppData\Local\hatch\env\virtual\sknnr\BxwCyoTQ\test\Lib\site-packages\sklearn\base.py", line 800, in __sklearn_tags__
    tags = super().__sklearn_tags__()
           ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'super' object has no attribute '__sklearn_tags__'
```
so I'm clearly pretty confused by this new tag system.)

Reply:
> I see that previously in GNNRegressor, we had _more_tags return multioutput=True, but that seems to be different from the new __sklearn_tags__ defining single_output=False?

From my reading of [the docs](https://scikit-learn.org/dev/modules/generated/sklearn.utils.TargetTags.html), I *believe* `multi_output=True` (which we now inherit, I think from `KNeighborsRegressor`) and `single_output=False` would mean that our estimators supports *only* multi-output. 

I'm less sure on the `two_d_labels` tag, which is defined as "Whether the input is a 2D labels (y)". Given that there's a corresponding `one_d_labels`, my assumption is that it's similar to (and maybe partially redundant with) `multi_output`. I suppose the distinction is that a single-output estimator could still require 2D labels, i.e. if it reduced multiple labels to a single output.

> Then I thought for sure TransformerMixin would have an estimator_type of transformer, but I get this:

Interesting, I would have thought the same thing! Looking at the [tags for `StandardScaler`](https://github.com/scikit-learn/scikit-learn/blob/fba028b07ed2b4e52dd3719dad0d990837bde28c/sklearn/preprocessing/_data.py#L1130), most are inherited from `BaseEstimator` including `estimator_type=None`. I get the sense that the tag system is still a work in progress, and really isn't widely used outside of their testing framework, so maybe that missing tag doesn't have any practical impact. I imagine it wouldn't hurt to define it ourselves, though.

> But I do agree with you that those seem to be new tags that weren't in the previous _more_tags.

Thanks for confirming! I'll go ahead and revert those.

Reply:
@aazuspan, circling back to this.  I had talked about creating a separate PR to require 2-D `y` data for `MSNRegressor` and `CCorATransformer`.  I started looking into this and came across [this comment](https://github.com/lemma-osu/sknnr/pull/44#issuecomment-1688726979) where I had convinced myself that 1-D `y` data was OK following the lead of `sklearn.cross_decomposition.CCA`.  I wanted to get your opinion before proceeding.

As I understand canonical correlation analysis, it has `X (n x p)` and `Y (n x r)` matrices.  The goal is to find coefficients for `X` and `Y` that produce linear combinations `X'` and `Y'` where `X'` and `Y'` are maximally correlated.  When I was thinking about a 1-D `Y` matrix (i.e. `r=1`), my assumption was that the user was using this in place of `Y'`, i.e. they had already found the coefficients that maximized the correlation, but only on the `Y` side.  But that doesn't really make a lot of sense.  What I failed to consider is that `Y` could be a single "species" and therefore the goal would be to find the coefficients in `X` that maximized correlation with that single species.

When looking back at the original [Moeur and Stage article](https://academic.oup.com/forestscience/article/41/2/337/4626909), it seems clear to me that their intention was that `r > 1` (at least their example data has `r=11`), but I guess the point is that *technically* `CCorA` still works if `r=1`.  (I'm honestly not totally clear if `CCA` would work similarly and still need to verify this). I guess at this point I'm leaning toward leaving the code as is, allowing for 1-D `y` arrays.

Any thoughts?  (I know this is getting buried into a PR that has no relevance to this topic, so perhaps hoisting it to a new issue makes sense.)

Reply:
I'm pretty far out of my depth here, but I think I'm *mostly* following your explanation - thanks for walking me through it.

> I guess at this point I'm leaning toward leaving the code as is, allowing for 1-D y arrays.

I agree with your conclusion here. It sounds like 1) `CCorA` and `MSN` technically work with with 1D data and 2) there are valid reasons to use 1D data (your single species example), so it makes sense to me to give users that tool and let them decide if it's appropriate for their data. There's definitely a limit to that rationale where a feature turns into a footgun, but it doesn't seem like that's where this falls.

Out of curiosity, do you know how yaImpute handles this?

Reply:
> Out of curiosity, do you know how yaImpute handles this?

It appears to succeed without warning when I subset the `y` data down to a single column.  So that seems like another reason to keep with the status quo.

Which gets us back to the original intent of your PR :smile:.  Do you think it makes sense to add those additional tags to `GNNRegressor` and `CCATransformer` now that I've set the fire alarm off unnessarily?!  Probably in a new PR?

Reply:
Good to know! 

Yeah, I think that's probably a good idea. I'll open a new issue just so we can consolidate some discussion and think through any other unsettled tag questions.